android/src/main/java/com/jhomlala/better_player/BetterPlayer.java

```diff
@@ -122,6 +125,71 @@
         setupVideoPlayer(eventChannel, textureEntry, result);
     }
 
+    static void preCache(Context context, String key, String dataSource, long preCacheSize, long maxCacheSize, long maxCacheFileSize, Map<String, String> headers, Result result) {
+
+        Uri uri = Uri.parse(dataSource);
+        DataSource.Factory dataSourceFactory;
+
+        String userAgent = System.getProperty(USER_AGENT_PROPERTY);
```

Comment:
We have same code in normal player. Can you create some common method so there won't be code duplication?